flutter_example/lib/main.gen.dart

```diff
@@ -114,3 +92,42 @@ class SimpleDisposable implements Disposable {
     print('disposing $this');
   }
 }
+
+// TODO(goderbauer): This could also be generated by a macro.
+class MyColoredFill extends LeafRenderObjectWidget {
+  const MyColoredFill({Key? key, required this.color}) : super(key: key);
+
+  final Color color;
+
+  @override
+  RenderObject createRenderObject(BuildContext context) {
+    return MyColoredFillRenderBox(color: color);
+  }
+
+  @override
+  void updateRenderObject(BuildContext context,  MyColoredFillRenderBox renderObject) {
+    renderObject.color = color;
+  }
+}
+
+class MyColoredFillRenderBox extends RenderBox {
+  MyColoredFillRenderBox({required Color color}) : _color = color;
+
+  @RenderAccessors(needsPaint: true) Color _color;
```

Comment:
I wish this annotation could also somehow add the color param and the initializer to the constructor, but didn't see a way to do that.

Reply:
That would need to be a separate macro (a class declaration macro most likely) - you can see the data class example in the dart vm examples.